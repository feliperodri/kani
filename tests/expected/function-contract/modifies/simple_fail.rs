// Copyright Kani Contributors
// SPDX-License-Identifier: Apache-2.0 OR MIT
// kani-flags: -Zfunction-contracts

#[kani::requires(*ptr < 100)]
fn modify(ptr: &mut u32) {
    *ptr += 1;
}

#[kani::proof_for_contract(modify)]
fn main() {
    let _ = Box::new(());
    let mut i = kani::any();
    modify(&mut i);
}
