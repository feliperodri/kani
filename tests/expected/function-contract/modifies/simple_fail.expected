assigns\
- Status: FAILURE\
- Description: "Check that *ptr is assignable"

Failed Checks: Check that *ptr is assignable

VERIFICATION:- FAILED
